src/main/java/mezz/jei/ingredients/IngredientManager.java

```diff
@@ -45,10 +49,9 @@ public IngredientManager(
 		}
 		this.ingredientsMap = ingredientsMapBuilder.build();
 
-		//noinspection UnstableApiUsage
 		this.registeredIngredientTypes = ingredientInfos.stream()
 			.map(IngredientInfo::getIngredientType)
-			.collect(ImmutableList.toImmutableList());
+			.collect(Collectors.toList());
```

Comment:
Shouldn't this be `.toList()` instead of `.collect(Collectors.toList())` to maintain immutability of the list?

Reply:
I have some weird cases where the stream types become complicated because of generics.

One example:
Give these types:
```java
// List<ITypedIngredient<?>> matches;
// List<Optional<ITypedIngredient<?>>> displayIngredients;
```

this way
```java
displayIngredients = matches.stream()
				.map(Optional::of)
				.collect(Collectors.toList());
```
works,

but if I use `.toList()` it needs some explicit casting:
```java
displayIngredients = matches.stream()
				.<Optional<ITypedIngredient<?>>>map(Optional::of)
				.toList();
```

I'll see if I can figure out the casting on more of these because it seems beneficial to make them properly immutable.